x-pack/plugins/security_solution/public/management/components/endpoint_responder/hooks.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import { useIsMounted } from '../../hooks/use_is_mounted';
+import { useGetActionDetails } from '../../hooks/endpoint/use_get_action_details';
+import { ACTION_DETAILS_REFRESH_INTERVAL } from './constants';
+import type { ActionRequest, ActionRequestComponentProps } from './types';
+import type { useSendIsolateEndpointRequest } from '../../hooks/endpoint/use_send_isolate_endpoint_request';
+import type { useSendReleaseEndpointRequest } from '../../hooks/endpoint/use_send_release_endpoint_request';
+
+export const useUpdateActionState = ({
+  actionRequestApi,
+  actionRequest,
+  command,
+  endpointId,
+  setStatus,
+  setStore,
+  isPending,
+}: Pick<ActionRequestComponentProps, 'command' | 'setStatus' | 'setStore'> & {
+  actionRequestApi: ReturnType<
+    typeof useSendIsolateEndpointRequest | typeof useSendReleaseEndpointRequest
+  >;
+  actionRequest?: ActionRequest;
+  endpointId?: string;
+  isPending: boolean;
+}) => {
+  const isMounted = useIsMounted();
+  const actionRequestSent = Boolean(actionRequest?.requestSent);
+  const { data: actionDetails } = useGetActionDetails(actionRequest?.actionId ?? '-', {
+    enabled: Boolean(actionRequest?.actionId) && isPending,
+    refetchInterval: isPending ? ACTION_DETAILS_REFRESH_INTERVAL : false,
+  });
+
+  // Send Isolate request if not yet done
+  useEffect(() => {
+    if (!actionRequestSent && endpointId) {
+      const request: ActionRequest = {
+        requestSent: true,
+        actionId: undefined,
+      };
+
+      actionRequestApi
+        .mutateAsync({
+          endpoint_ids: [endpointId],
+          comment: command.args.args?.comment?.[0],
+        })
+        .then((response) => {
+          request.actionId = response.action;
+
+          if (isMounted) {
+            setStore((prevState) => {
+              return { ...prevState, actionRequest: request };
+            });
+          }
+        })
+        .finally(() => {
+          setStore((prevState) => {
```

Comment:
This assume a successful api call and that the component is still mounted. but also - It seems redundant because you already have the `.then()` above that does this. Am I missing something?

Reply:
but this is already done in the `.then()` block, so why do it again?

Reply:
b5638ad1fea298b34708515eee44e80504e436c0